src/renderer/themes.ts

```diff
@@ -53,6 +55,47 @@ export async function readThemeFile(
   }
 }
 
+/**
+ * Create theme file and show in folder.
+ *
+ * @param {FiddleTheme} theme
+ * @param {string} [name]
+ * @returns {Promise<LoadedFiddleTheme>}
+ */
+export async function createThemeFile(
+  theme: FiddleTheme | LoadedFiddleTheme,
+  name?: string,
+): Promise<LoadedFiddleTheme> {
+  // Filter out file and css keys if they exist
+  theme = Object.fromEntries(
+    Object.entries(theme).filter(([key]) => !['file', 'css'].includes(key)),
+  ) as FiddleTheme;
```

Comment:
```suggestion
  const { file: _file, css: _css, ...result } = theme;
  theme = result as FiddleTheme;
```

Reply:
That was my original approach, but it isn't applicable in this code base for two reasons:
* Fails the ESLint `@typescript-eslint/no-unused-vars` check
  * Although `.eslintrc.js` could be tweaked to use `varsIgnorePattern` to allow those
* Fails type checking: `Property 'file' does not exist on type 'LoadedFiddleTheme | FiddleTheme'. ts(2339)`

As such, I think the current code is the "least bad" solution, even though it's a bit ugly. Fighting TypeScript in this case can be a headache.

Reply:
> Fails the ESLint @typescript-eslint/no-unused-vars check 

```json
"@typescript-eslint/no-unused-vars": ["error", {
  "argsIgnorePattern": "^_",
}]
```

> Fails type checking: Property 'file' does not exist on type 'LoadedFiddleTheme | FiddleTheme'. ts(2339)

There are three solutions to this.

#### 1.

add `// @ts-ignore`.

```ts
// @ts-ignore
const { file: _file, css: _css, ...result } = theme;
```

#### 2.

add `as`

```ts
const { file: _file, css: _css, ...result } = theme as LoadedFiddleTheme & FiddleTheme;
```

#### 3.

add `OptionalMerge` type

```ts
type OptionalMerge<T1, T2> = {
    [K in keyof (T1 & T2)]: K extends keyof T1
        ? K extends keyof T2
            ? T1[K] | T2[K]
            : T1[K] | undefined
        : K extends keyof T2
            ? T2[K] | undefined
            : never
};

const { file: _file, css: _css, ...result } = theme as OptionalMerge<LoadedFiddleTheme, FiddleTheme>;
```

See: [typescriptlang-play](https://www.typescriptlang.org/play?#code/C4TwDgpgBA8mwEsD2A7AhgGwLIQE4HMIAeAFQEYAaKEgJgD4oBeKAbwCgpOoBtAaSgQooAawggkAMygAKclABk1GgEoAugC4o-CAA9gEFABMAziLGTqZDlxsB+LVF36jp0eKm1rN71Hvk+qlAAPkoBXj5cmv68gSEArkYQEoIQhuE2mtp6BiZm7krpEX40AcFQCYZJKWkR3pooEABueGwAvgDcbGygkFAAgkys4ckYEJrGwLiC+OEAxsbG45PTw0hIS1MoM61dPdAAQoPsNvOLUBObMzYSaxsrNgBGaLh3W21ds6gTUMAAFhAAW2gzBYrSgaFMAxC+06bE+KG+LCgIzGUAA+iiqKdNGjTlQAHSE3AQYxxDDAKCtZh-QHQCGweDIdDYPCEIh9Kj7OjtIA)

Reply:
Personally, I recommend using priority is: 3 > 2 > 1

Reply:
> add `// @ts-ignore`.

I don't consider this a valid solution, personally. 🙂

> add `as`

Since `LoadedFiddleTheme` extends `FiddleTheme`, no need for `as LoadedFiddleTheme & FiddleTheme`, just `as LoadedFiddleTheme`. Either way I don't like this option as the typing is then incorrect - destructuring the `name` property would have type `string` even though it may actually be `undefined` at run-time.

> add `OptionalMerge` type

This would work, but personally I don't think it's necessary or worth the tradeoff. `OptionalMerge` is not very readable code (that's just TypeScript types like that in general, they're higher cognitive load to parse). I think the existing code is fine, and doesn't require introducing `OptionalMerge` and changing the ESLint configuration.